src/server_manager/web_app/ui_components/outline-gcp-create-server-app.ts

```diff
@@ -0,0 +1,341 @@
+  'us-central1': `${FLAG_IMAGE_DIR}/us.png`,
+  'us-east1': `${FLAG_IMAGE_DIR}/us.png`,
+  'us-east4': `${FLAG_IMAGE_DIR}/us.png`,
+  'us-west1': `${FLAG_IMAGE_DIR}/us.png`,
+  'us-west2': `${FLAG_IMAGE_DIR}/us.png`,
+  'us-west3': `${FLAG_IMAGE_DIR}/us.png`,
+  'us-west4': `${FLAG_IMAGE_DIR}/us.png`,
+};
+
+// TODO: Handle network and authentication errors
+
+@customElement('outline-gcp-create-server-app')
+export class GcpCreateServerApp extends LitElement {
+  @property({type: Function}) localize: Function;
+  @internalProperty() private currentPage = '';
+  @internalProperty() private selectedProjectId = '';
+  @internalProperty() private selectedBillingAccountId = '';
+
+  private account: GcpAccount;
+  private project: Project;
+  private billingAccounts: BillingAccount[] = [];
+  private regionPicker: OutlineRegionPicker;
+
+  static get styles() {
+    return [
+      COMMON_STYLES, css`
+      .container {
+        display: flex;
+        flex-direction: column;
+        justify-content: center;
+        height: 100%;
+        align-items: center;
+        padding: 132px 0;
+        font-size: 14px;
+      }
+      .card {
+        display: flex;
+        flex-direction: column;
+        align-items: stretch;
+        justify-content: space-between;
+        margin: 24px 0;
+        padding: 24px;
+        background: var(--background-contrast-color);
+        box-shadow: 0 0 2px 0 rgba(0, 0, 0, 0.14), 0 2px 2px 0 rgba(0, 0, 0, 0.12), 0 1px 3px 0 rgba(0, 0, 0, 0.2);
+        border-radius: 2px;
+      }
+      .section {
+        padding: 24px 12px;
+        background: var(--background-contrast-color);
+        border-radius: 2px;
+      }
+      .section:not(:first-child) {
+        margin-top: 8px;
+      }
+      .section-header {
+        padding: 0 6px 0;
+        display: flex;
+      }
+      .instructions {
+        font-size: 16px;
+        line-height: 26px;
+        margin-left: 16px;
+        flex: 2;
+      }
+      .stepcircle {
+        height: 26px;
+        width: 26px;
+        font-size: 14px;
+        border-radius: 50%;
+        float: left;
+        vertical-align: middle;
+        color: #000;
+        background-color: #fff;
+        margin: auto;
+        text-align: center;
+        line-height: 26px;
+      }
+      @media (min-width: 1025px) {
+        paper-card {
+          /* Set min with for the paper-card to grow responsively. */
+          min-width: 600px;
+        }
+      }
+      .card p {
+        color: var(--light-gray);
+        width: 100%;
+        text-align: center;
+      }
+      paper-button {
+        color: var(--light-gray);
+        width: 100%;
+        border: 1px solid rgba(255, 255, 255, 0.12);
+        border-radius: 2px;
+      }
+      paper-button[disabled] {
+        color: var(--medium-gray);
+        background: transparent;
+      }`
+    ];
+  }
+
+  render() {
+    return html`
+      <iron-pages id="pages" attr-for-selected="id" .selected="${this.currentPage}">
```

Comment:
There's no need for iron pages with LitElement. Use a conditional block to output only the html you need.

Reply:
Perhaps you have a `renderStep` variable that holds the render method to use.

Reply:
You may benefit from having state machine objects, each with their own state, and render() method.
Then you can do things like:
this.state = setupBillingPage()

Reply:
I added the conditional logic to output only the html for the given step.